### PR TITLE
[Enhancement] Avoid high IOPS in orc reader when datacache enabled

### PR DESCRIPTION
**Pull request by Smith-Cruise**
Why I'm doing:
ORC's iops is high when datacache enabled

What I'm doing:
Just reduce it. And improve profile.

image

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5

### CODE REVIEW

File: be/src/exec/hdfs_scanner.h [hunk 1]
```diff
@@ -77,7 +77,7 @@ struct HdfsScanStats {
     int64_t group_min_round_cost = 0;
 
     // ORC only!
```
Comment (dirtysalt) [inline be/src/exec/hdfs_scanner.h:79]: then remove it.

File: be/src/exec/hdfs_scanner_orc.cpp [hunk 2]
```diff
@@ -548,26 +548,33 @@ Status HdfsOrcScanner::do_init(RuntimeState* runtime_state, const HdfsScannerPar
...
-                                                   kORCProfileSectionPrefix);
-    stripe_number_counter = ADD_CHILD_COUNTER(root, "StripeNumber", TUnit::UNIT, kORCProfileSectionPrefix);
-    for (auto v : _app_stats.stripe_sizes) {
-        COUNTER_UPDATE(stripe_sizes_counter, v);
+    double total_stripe_size = 0;
+    for (const auto& v : _app_stats.orc_stripe_sizes) {
+        total_stripe_size += v;
     }
-    COUNTER_UPDATE(stripe_number_counter, _app_stats.stripe_sizes.size());
+    double avg_stripe_size = 0;
```
Comment (dirtysalt) [inline be/src/exec/hdfs_scanner_orc.cpp:563]: what's the final profile looks like? Can you post a profile image?

File: be/src/formats/orc/apache-orc/c++/src/Reader.cc [hunk 3]
```diff
@@ -1016,7 +1016,8 @@ void RowReaderImpl::buildIORanges(std::vector<InputStream::IORange>* io_ranges)
     for (const proto::Stream& stream : currentStripeFooter.streams()) {
         uint32_t columnId = stream.column();
         uint64_t length = stream.length();
-        if (selectedColumns[columnId] || lazyLoadColumns[columnId]) {
+        // ColumnId = 0 is root column, we always need it
+        if (columnId == 0 || selectedColumns[columnId] || lazyLoadColumns[columnId]) {
             io_ranges->emplace_back(InputStream::IORange{.offset = offset, .size = length});
```
Comment (dirtysalt) [inline be/src/formats/orc/apache-orc/c++/src/Reader.cc:1021]: is this a bug?

Comment (Smith-Cruise) [reply be/src/formats/orc/apache-orc/c++/src/Reader.cc:1021]: image

ColumnId = 0 is root node, must always include it.

File: be/src/exec/hdfs_scanner_orc.cpp (hunk 2 above)
Comment (Smith-Cruise) [reply be/src/exec/hdfs_scanner_orc.cpp:563]: done

File: be/src/exec/hdfs_scanner.h (hunk 1 above)
Comment (Smith-Cruise) [reply be/src/exec/hdfs_scanner.h:79]: done

Comment (Smith-Cruise) [reply be/src/exec/hdfs_scanner.h]: done

**Review by letian-jiang — APPROVED** [review]

**Review by dirtysalt — DISMISSED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Pull request merged**